plugins/permission-common/src/PermissionClient.ts

```diff
@@ -140,33 +199,21 @@ export class PermissionClient implements PermissionAuthorizer {
     }
 
     const responseBody = await response.json();
-    this.assertValidResponse(request, responseBody);
 
-    const responsesById = responseBody.items.reduce((acc, r) => {
+    const parsedResponse = responseSchema(
+      itemSchema,
+      new Set(request.items.map(({ id }) => id)),
+    ).parse(responseBody);
+
+    const responsesById = parsedResponse.items.reduce((acc, r) => {
       acc[r.id] = r;
       return acc;
-    }, {} as Record<string, IdentifiedPermissionMessage<EvaluatePermissionResponse>>);
+    }, {} as Record<string, typeof itemSchema._type>);
```

Comment:
Just matching the [recommended approach from the docs](https://github.com/colinhacks/zod#type-inference)

```suggestion
    }, {} as Record<string, z.infer<typeof itemSchema>>);
```